RSAKeyVaultProvider/RSAKeyVault.cs

```diff
@@ -23,16 +23,21 @@ public RSAKeyVault(KeyVaultContext context)
 
         public override byte[] SignHash(byte[] hash, HashAlgorithmName hashAlgorithm, RSASignaturePadding padding)
         {
+            // Key Vault's API is known to use CA(false) everywhere. This should not deadlock.
+            return SignHashAsync(hash, hashAlgorithm, padding).GetAwaiter().GetResult();
+        }
```

Comment:
Not sure... it wasn't there originally. I just extracted the async parts and made them public.